backend/src/case_file/case_file.service.ts

```diff
@@ -906,8 +1027,435 @@ export class CaseFileService {
+            );
+            throw new Error(
+              "Failed to update equipment geometry due to a database error."
+            );
+          }
+
+        // Check for updated or added actions
+        const actions = equipmentRecord.actions;
+        for (const action of actions) {
+          if (action.actionGuid) {
+            this.logger.debug(
+              `Updating equipment action: ${JSON.stringify(action)}`
+            );
+            // If actionGuid exists, it means the action already exists and needs to be updated
+            await db.action.update({
+              where: { action_guid: action.actionGuid },
+              data: {
+                action_date: action.date,
+                actor_guid: action.actor,
+                update_utc_timestamp: new Date(),
+              },
+            });
+          } else {
+            // we're adding a new action, so find the action type action xref needed for this
+            this.logger.debug(
+              `Creating new equipment action: ${JSON.stringify(action)}`
+            );
+            let actionTypeActionXref =
+              await db.action_type_action_xref.findFirstOrThrow({
+                where: {
+                  action_type_code: ACTION_TYPE_CODES.EQUIPMENT,
+                  action_code: action.actionCode,
+                },
+                select: {
+                  action_type_action_xref_guid: true,
+                },
+              });
+
+            this.logger.debug(`Found action xref`);
+            const caseFileGuid = caseFile.caseIdentifier;
+            // create the action records (this may either be setting an equipment or removing an equipment)
+            const data = {
+              case_guid: caseFileGuid,
+              action_type_action_xref_guid:
+                actionTypeActionXref.action_type_action_xref_guid,
+              actor_guid: action.actor,
+              action_date: action.date,
+              active_ind: action.activeIndicator,
+              create_user_id: updateEquipmentInput.updateUserId,
+              create_utc_timestamp: new Date(),
+              equipment_guid: equipmentGuid,
+            };
+
+            this.logger.debug(
+              `Adding new equipment action as part of an update: ${JSON.stringify(
+                data
+              )}`
+            );
+
+            await db.action.create({
+              data: data,
+            });
+          }
+        }
+      });
+
+      const caseFileGuid = caseFile.caseIdentifier;
+      caseFileOutput = await this.findOne(caseFileGuid);
+    } catch (error) {
+      this.logger.error("An error occurred during equipment update:", error);
+      throw new GraphQLError(
+        "An error occurred during equipment update. See server log for details",
+        error
+      );
+    }
+    return caseFileOutput;
+  }
+
+  // create an equipment record - with actions to either set the equipment, or set and remove the equipment
+  async createEquipment(
```

Comment:
@afwilcox while this does work, I think this could use some additional TLC. there's a couple things here that rub me the wrong way but not enough to prevent us from merging this story. Equipment being an array, and being hardcoded to the first element is problematic. This should be addressed in a refactor.